app/actions/user.js

```diff
@@ -1,8 +1,79 @@
+import{
+  AsyncStorage
+}from 'react-native';
 import * as types from './types';
+const BreakoutApi= require("breakout-api-client");
 
 export function setUser(user){
   return{
     type: types.SET_USER,
     payload:user
   }
 }
+
+export function setUsername(username){
+  return {
+    type: types.SET_USERNAME,
+    payload:username
+  }
+}
+
+export function setLoggingIn(logging){
+  return {
+    type: types.LOGGING_IN,
+    payload:logging
+  }
+}
+
+export function login(username,password){
+  return (dispatch,getState) => {
+    console.log(username+" "+password);
+    let curState = getState()
+    let loggingIn = curState.isLoggingIn;
+    console.log(BreakoutApi);
+    dispatch(setLoggingIn(true));
+    sendLogin(username,password).then((resp)=>{
+      if(resp.success === 1){
+        console.log("success");
+        saveUserData(username,password).then(()=>{
+          dispatch(setUser(resp.account));
+        });
+      } else {
+        console.log("failure");
```

Comment:
Make error messages more expressive in order to allow easier debugging! E.g. "Failure during login: REASON"

Reply:
Done, coming with the next commit

Reply:
Done, coming with the next commit